src/server/components/dashboard/controllers.ts

```diff
@@ -349,3 +365,114 @@ export async function feedbackController(
     next(error);
   }
 }
+
+export async function listsEditAnnualReviewDateController(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  try {
+    const { listId } = req.params;
+    const list = await findListById(listId);
+    const annualReviewStartDate = formatAnnualReviewDate((list as List), "annualReviewStartDate");
+
+    res.render("dashboard/lists-edit-annual-review-date", {
+      ...DEFAULT_VIEW_PROPS,
+      error: {
+        text: req.flash('annualReviewError')[0],
+        href: "#annualReviewDateForm",
+      },
+      annualReviewStartDate,
+      list,
+      csrfToken: getCSRFToken(req),
+    });
+  } catch(error) {
+    logger.error(`listsEditAnnualReviewDateController Error: ${(error as Error).message}`)
+    next(error);
+  }
+}
+
+export async function listsEditAnnualReviewDatePostController(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  try {
+
+    return (req.body.action === "confirmNewDate")
+      ? await confirmNewAnnualReviewDate(req, res)
+      : await updateNewAnnualReviewDate(req, res);
+
+  } catch(error) {
+    logger.error(`listsEditAnnualReviewDatePostController Error: ${(error as Error).message}`)
+    next(error);
+  }
+}
+
+async function confirmNewAnnualReviewDate(req: Request, res: Response): Promise<void> {
+  const todaysDate = new Date();
+
+  const { listId } = req.params;
+  const list = await findListById(listId);
+  let annualReviewYear = todaysDate.getFullYear();
+  let maxDate = add(todaysDate, { months: 6 });
+
+  const {month, day} = req.body;
+  const { annualReviewStartDate } = (list as List).jsonData;
+
+  if (annualReviewStartDate) {
+    const annualReviewDate = new Date(annualReviewStartDate);
+    annualReviewYear = annualReviewDate.getFullYear();
+    maxDate = add(annualReviewDate, { months: 6 });
+  }
+
+  let parsedDate = parse(`${month}/${day}/${annualReviewYear}`, "P", new Date());
+
+  if (isPast(parsedDate)) {
+    parsedDate = add(parsedDate, { years: 1 });
+  }
+
+  if (!month || !day) {
+    req.flash('annualReviewError', 'Enter a date for the annual review');
+    return res.redirect(`${dashboardRoutes.listsEditAnnualReviewDate.replace(":listId", listId)}`);
+  }
+
+  if ((month === "2" && day === "29") || !isValid(parsedDate)) {
```

Comment:
general note on use of literals.  To provide a new dev with context around this, it probably makes sense to wrap `(month === "2" && day === "29")` in a helper function so the code is self documenting.